### PR TITLE
Incorrect statusPageUrl and healthCheckUrl are reported to Eureka if …

### PR DESCRIPTION
**Pull request by Aloren**
…management.context-path is not set

### CODE REVIEW

**Comment by spencergibb** [comment]
So doesn't contextPath affect management if management.port is not set?

**Comment by Aloren** [comment]
Forgot about that cases. Thanks. I will add handling of other cases as well.

File: spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java [hunk 1]
```diff
@@ -113,38 +113,46 @@ public EurekaClientConfigBean eurekaClientConfigBean() {
...
-		int managementPort = Integer.valueOf(propertyResolver.getProperty("management.port", String.valueOf(nonSecurePort)));
-		String managementContextPath = propertyResolver.getProperty("management.contextPath", propertyResolver.getProperty("server.contextPath", "/"));
+		Integer managementPort = propertyResolver.getProperty("management.port", Integer.class);// nullable. should be wrapped into optional
+		String managementContextPath = propertyResolver.getProperty("management.contextPath");// nullable. should be wrapped into optional
+
 		EurekaInstanceConfigBean instance = new EurekaInstanceConfigBean(inetUtils);
 		instance.setNonSecurePort(nonSecurePort);
 		instance.setInstanceId(getDefaultInstanceId(propertyResolver));
 		instance.setPreferIpAddress(preferIpAddress);
-		if (managementPort != nonSecurePort && managementPort != 0) {
```
Comment (Aloren) [inline spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java]: Why there was a check for managementPort != nonSecurePort?
Shouldn't we set urls if management port is equal to nonSecurePort?

**Comment by spencergibb** [comment]
Looks like a rebase is in order. Also, there should be a test case for management context path set, but not management port.

**Comment by Aloren** [comment]
Rebased to masted and added test

**Comment by ryanjbaxter** [comment]
@Aloren sorry but this needs another rebase.

**Comment by Aloren** [comment]
:) I've refactored my solution a bit to return `ManagementMetadata` and rebased. Please check also test cases I may have missed smth. Thanks!

**Pull request merged**